cryptographic_estimators/MREstimator/MRAlgorithms/bruteforce.py

```diff
@@ -0,0 +1,156 @@
+
+
+class BruteForce(MRAlgorithm):
+    r"""
+    Construct an instance of BruteForce estimator
+
+
+    INPUT:
+
+    - ``problem`` -- an instance of the MRProblem class
+    - ``w`` -- linear algebra constant (default: 3)
+    - ``theta`` -- exponent of the conversion factor (default: 2)
+
+    EXAMPLES::
+
+        sage: from cryptographic_estimators.MREstimator.MRAlgorithms.bruteforce import BruteForce
+        sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+        sage: E = BruteForce(MRProblem(q=7, m=9, n=10, k=15, r=4))
+        sage: E
+        BruteForce estimator for the MinRank problem with (q, m, n, k, r) = (7, 9, 10, 15, 4)
+    """
+
+    def __init__(self, problem: MRProblem, **kwargs):
+
+        super(BruteForce, self).__init__(problem, **kwargs)
+
+        q, m, n, k, r = self.problem.get_parameters()
+        self.set_parameter_ranges('a', 0, min(n - r, ceil(k / m)))
+        self.set_parameter_ranges('lv', 0, r)
+        self._name = "BruteForce"
+
+    @optimal_parameter
+    def a(self):
+        """
+        Return the optimal `a`, i.e. no. of vectors to guess in the kernel of the low-rank matrix
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.bruteforce import BruteForce
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: BFE = BruteForce(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: BFE.a()
+            1
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.bruteforce import BruteForce
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: BFE = BruteForce(MRProblem(q=16, m=15, n=15, k=78, r=6))
+            sage: BFE.a()
+            5
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_KERNEL_VECTORS_TO_GUESS)
+
+    @optimal_parameter
+    def lv(self):
+        """
+        Return the optimal `lv`, i.e. no. of entries to guess in the solution
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.bruteforce import BruteForce
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: BFE = BruteForce(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: BFE.lv()
+            0
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.bruteforce import BruteForce
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: BFE = BruteForce(MRProblem(q=16, m=15, n=15, k=78, r=6))
+            sage: BFE.lv()
+            0
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_COEFFICIENTS_TO_GUESS)
+
+    def _BFE_time_complexity_helper_(self, q, k_reduced, r):
+        time = 0
```

Comment:
Although thats an internal function: maybe its beneficial if it had a doc string and types at the arguments.

Reply:
I agree, either it should have a descriptive name or a docstring